### PR TITLE
sanitycheck: fixed handling of retries

### LINKED ISSUE #13956: CI scripting doesn't retry modifications to tests on non-default platforms
It seems like CI gets it's multipass/retry behavior confused in the presence of changes to test code.  When a test is modified, the "--save-tests" pre-pass uses --all to enable all platforms to make sure the test at least builds everywhere.  But the final "--only-failed" pass seems not to use that list and instead just retries the default list, so any tests that fail the first time never actually get retried, and the test as a whole shows a failure.

PR #13916 showed a situation like this, note the single (spurious) failure in the first run and the fact that zero tests got run in the retry:

```
+ /home/buildslave/src/github.com/zephyrproject-rtos/zephyr/scripts/sanitycheck --inline-logs --enable-coverage -N --load-tests test_file.txt --subset 1/5
JOBS: 12
Running only a subset: 1/5
Cleaning output directory /home/buildslave/src/github.com/zephyrproject-rtos/zephyr/sanity-out
445 tests selected, 0 tests discarded due to filters
  1/443 adafruit_trinket_m0       samples/synchronization/sample.synchronization     PASSED
...
...
...
443/443 native_posix              tests/net/tx_timestamp/net.timestamp               PASSED
442 of 443 tests passed with 0 warnings in 1263 seconds
+ sleep 10
+ /home/buildslave/src/github.com/zephyrproject-rtos/zephyr/scripts/sanitycheck --inline-logs --enable-coverage -N --only-failed --outdir=out-2nd-pass
JOBS: 12
Selecting default platforms per test case
Building testcase defconfigs...
0 tests selected, 91523 tests discarded due to filters
0 of 0 tests passed with 0 warnings in 2 seconds
+ rm -f test_file.txt
```

### PR DESCRIPTION
**Pull request by nashif**
When re-running failed tests, do not go through any filtering, just load
the failed tests directly and execute them. The filtering was done based on
default platforms and any tests that were failing on non-default platforms
were ignored.

Fixes #13956

### CODE REVIEW

File: scripts/sanitycheck [hunk 1]
```diff
@@ -3237,6 +3229,8 @@ def main():
     discards = []
     if options.load_tests:
         ts.load_from_file(options.load_tests)
+    elif options.only_failed:
+        ts.get_last_failed()
```
Comment (andyross) [inline scripts/sanitycheck]: OMG, this is line three thousand two hundred and thirty three.

**Review by andyross — APPROVED** [review]
Makes sense, with a little horrified snark at the size of this script.

File: scripts/sanitycheck (hunk 1 above)
Comment (nashif) [reply scripts/sanitycheck]: :)

Comment (nashif) [reply scripts/sanitycheck]: yeah, we need to refactor the whole thing at some point, things have been adding up.

**Pull request merged**